### PR TITLE
Make StartInfo_TextFile_ShellExecute test more reliable

### LINKED ISSUE #25823: StartInfo_TextFile_ShellExecute  test failing for spanish windows (Non English CI)
```
Could not start C:\\Users\\runner\\AppData\\Local\\Temp\\ProcessStartInfoTests_q0umvtb5.dh2\\StartInfo_TextFile_ShellExecute_994_2372eb4f.txt UseShellExecute=True
Association details for '.txt'
------------------------------
Open command: C:\\Windows\\system32\\NOTEPAD.EXE %1
ProgID: txtfile

Expected: True
Actual:   False
```

### PR DESCRIPTION
**Pull request by adamsitnik**
**edit:**  introduce a new, unique extension and use the registry to associate it with Notepad to be always sure that it's going to be used

Fixes #25823

### CODE REVIEW

**Comment by adamsitnik** [comment]
@Anipik @ViktorHofer how can I trigger the "Spanish" CI?

File: src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 1]
```diff
@@ -956,26 +956,26 @@ public void StartInfo_NotepadWithContent(bool useShellExecute)
...
-                return; // [ActiveIssue(https://github.com/dotnet/runtime/issues/25823)]
-
             string tempFile = GetTestFilePath() + ".txt";
             File.WriteAllText(tempFile, $"StartInfo_TextFile_ShellExecute");
 
             ProcessStartInfo info = new ProcessStartInfo
             {
                 UseShellExecute = true,
-                FileName = tempFile,
+                FileName = "notepad.exe",
```
Comment (jkotas) [inline src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:972]: This change makes this test redundant. It is not testing what it was meant to test anymore.

**Comment by jkotas** [comment]
> otherwise, things like notepad++ get opened (common on dev machines) and the test fails

We should move the test to outer loop. It is ok for outer loop tests to have environment dependencies like having notepad.exe registered with .txt files.

File: src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: @jkotas great catch (as usual)! I did the [homework](https://stackoverflow.com/a/5255335/5852046)

**Comment by adamsitnik** [comment]
> We should move the test to outer loop. 

It's already an outer loop test:

https://github.com/dotnet/runtime/blob/38c9b98eb12a7cb976c77e89a070fa2803abc9df/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs#L962

> It is ok for outer loop tests to have environment dependencies like having notepad.exe registered with .txt files.

I think that we should introduce a new, unique extension and use the registry to associate it with Notepad to be always sure that it's going to be used. PTAL at my most recent commit

**Comment by ViktorHofer** [comment]
> @Anipik @ViktorHofer how can I trigger the "Spanish" CI?

Replied offline but to summarize, the "Send to helix" log shows which queues work was sent to:

```
Sending Job to Windows.81.Amd64.Open... 

 Sent Helix Job; see work items at https://helix.dot.net/api/jobs/547b1787-d975-4927-8d42-be6f276c4663/workitems?api-version=2019-06-17 

 Sending Job to Windows.10.Amd64.Server19H1.ES.Open... 

 Sent Helix Job; see work items at https://helix.dot.net/api/jobs/c034fbf8-a38e-48d6-acda-47ff9fe8f4dd/workitems?api-version=2019-06-17 

 Sending Job to (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504... 

 Sent Helix Job; see work items at https://helix.dot.net/api/jobs/2bc9eedd-d9e5-4b3c-8680-f70681e0ceea/workitems?api-version=2019-06-17 

 Waiting for completion of job 2bc9eedd-d9e5-4b3c-8680-f70681e0ceea on (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504 

 Waiting for completion of job c034fbf8-a38e-48d6-acda-47ff9fe8f4dd on Windows.10.Amd64.Server19H1.ES.Open 

 Waiting for completion of job 547b1787-d975-4927-8d42-be6f276c4663 on Windows.81.Amd64.Open
```

In this case, `Windows.10.Amd64.Server19H1.ES.Open` is the Spanish queue which already runs per PR.

**Comment by adamsitnik** [comment]
From the CI logs it looks like only `System.Net.Http.Functional.Tests.TelemetryTest_Http11.EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue` has failed (not related) and the rest have passed

**Comment by danmosemsft** [comment]
Thanks, this seems like a good experiment to try. Was there any evidence though that the txt mapping was missing/bad? IIRC we log the reg key and it always seemed correct, the launch still failed, which left me confused.

I've tried to stabilize this poor test for a long time.

**Comment by adamsitnik** [comment]
> Was there any evidence though that the txt mapping was missing/bad? 

No, but it was the only reason why it was not passing on my dev machine so I thought that it would be good to improve it.

@danmosemsft is it OK if I re-enable now and when it fails get back to it with new logs?

**Comment by adamsitnik** [comment]
FWIW it's hard to act based on the old logs like https://github.com/dotnet/runtime/issues/22007#issuecomment-383766923 because they contain source code line number, and the file has been modified many times and the line number does not tell me much

**Comment by danmosemsft** [comment]
Happy with whatever approach seems best to you.

**Comment by adamsitnik** [comment]
> Happy with whatever approach seems best to you.

@danmosemsft Can I get approval from you so I can merge the PR?

**Review by danmoseley — APPROVED** [review]

**Pull request merged**